### PR TITLE
DEP: Rename device

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

- Deprecate `Device` and `device.py`
- Move `HappiItem` to `item.py`
- Establish `OphydItem` and `LCLSItem`
- Cut unused fields while we're at it

## Motivation and Context

- `Device` is too confusing with the conflicting names in `ophyd`
- We don't use most of these fields. They can be added later when a need arises.

## How Has This Been Tested?

Not yet, looking for feedback. Old tests still pass, now with warnings.

## Where Has This Been Documented?

Not yet, looking for feedback.

### CODE REVIEW

File: happi/item.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+logger = logging.getLogger(__name__)
+
+# For back-compat to <py3.7
+try:
+    from re import Pattern
+except ImportError:
+    from re import _pattern_type as Pattern
+
+
+class EntryInfo(object):
```
Comment (klauer) [inline happi/item.py]:
```suggestion
class EntryInfo:
```

File: happi/item.py [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+                                                               self.default)
+
+    def __copy__(self):
+        return EntryInfo(doc=self.doc, optional=self.optional,
+                         enforce=self.enforce, default=self.default)
+
+
+class _HappiItemBase:
+    def __init_subclass__(cls, **kargs):
+        # These attributes are used by device so can not be overwritten
```
Comment (klauer) [inline happi/item.py]:
```suggestion
        # These attributes are not to be overwritten by subclasses
```

File: happi/item.py [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+            else:
+                setattr(self, info.key, info.default)
+        # Handle additional information
+        self.extraneous = kwargs
+        if kwargs:
+            logger.debug('Additional information for %s was defined %s',
+                         self.name, ', '.join(self.extraneous))
+
+    @property
+    def info_names(self):
```
Comment (klauer) [inline happi/item.py]: We should move these into the `_HappiItem` `__init_subclass__`, as `info_names`, `mandatory_info` does not change.

Comment (ZLLentz) [reply happi/item.py]: Done

**Comment by ZLLentz** [comment]
Next thing to do is to go through all of our methods that include `device` in the name, deprecate them, and give an alternate method with `item` in the name

**Comment by ZLLentz** [comment]
I'm also going to be going through internal source code and swapping `device` for `item` wherever appropriate to boost readability

**Comment by hhslepicka** [comment]
That is a great idea, @ZLLentz

File: happi/device.py [hunk 4]
```diff
@@ -415,14 +52,28 @@ class Device(HappiItem):
...
                                     self.__class__.__name__,
                                     self.name,
                                     self.prefix,
                                     self.z)
+
+    @property
+    def screen(self):
+        warnings.warn("The 'screen' keyword is no longer used in Happi as it "
+                      "lacks specificity. Use one of detailed_screen, "
+                      "embedded_screen, or engineering screen instead")
```
Comment (hhslepicka) [inline happi/device.py]:
```suggestion
                      "embedded_screen, or engineering_screen instead")
```

File: happi/device.py [hunk 5]
```diff
@@ -415,14 +52,28 @@ class Device(HappiItem):
...
+    parent = EntryInfo('If the device is a component of another, '
+                       'enter the name', enforce=str)
     args = copy.copy(HappiItem.args)
     args.default = ['{{prefix}}']
     kwargs = copy.copy(HappiItem.kwargs)
     kwargs.default = {'name': '{{name}}'}
 
+    def __init__(self, *args, **kwargs):
+        warnings.warn("happi.device.Device is deprecated. Please use "
+                      "OphydItem or LCLSItem.")
```
Comment (hhslepicka) [inline happi/device.py]:
```suggestion
                      "OphydItem or LCLSItem.", DeprecationWarning)
```

**Pull request merged**